receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper.go

```diff
@@ -64,31 +72,21 @@ func (s *scraper) scrape(_ context.Context) (pdata.Metrics, error) {
 		return md, scrapererror.NewPartialScrapeError(err, metricsLen)
 	}
 
-	initializeCPUTimeMetric(metrics.AppendEmpty(), s.startTime, now, cpuTimes)
-	return md, nil
-}
-
-func initializeCPUTimeMetric(metric pdata.Metric, startTime, now pdata.Timestamp, cpuTimes []cpu.TimesStat) {
-	metadata.Metrics.SystemCPUTime.Init(metric)
-
-	ddps := metric.Sum().DataPoints()
-	ddps.EnsureCapacity(len(cpuTimes) * cpuStatesLen)
 	for _, cpuTime := range cpuTimes {
-		appendCPUTimeStateDataPoints(ddps, startTime, now, cpuTime)
+		appendCPUTimeStateDataPoints(s.mb, now, cpuTime)
 	}
+	s.mb.Emit(metrics)
+	return md, nil
 }
 
 const gopsCPUTotal string = "cpu-total"
 
-func initializeCPUTimeDataPoint(dataPoint pdata.NumberDataPoint, startTime, now pdata.Timestamp, cpuLabel string, stateLabel string, value float64) {
-	attributes := dataPoint.Attributes()
+func initializeCPUTimeDataPoint(mb *metadata.MetricsBuilder, now pdata.Timestamp, cpuLabel string, stateLabel string,
+	value float64) {
 	// ignore cpu attribute if reporting "total" cpu usage
+	cpuAttributeValue := pdata.NewAttributeValueEmpty()
 	if cpuLabel != gopsCPUTotal {
```

Comment:
Do we know when this happens?

Reply:
Yes, it's always reported on windows -- no per CPU metrics there. Also gopsutil has on option to not peport per CPU metrics, in that case this attribute is reported

Reply:
I don't believe you, see https://github.com/shirou/gopsutil/blob/a00b8eac527dd3e883ed6460ca1edc0915bbb6cb/v3/cpu/cpu_windows.go#L63 

I think this is reported only if "Times" func is called with perCpu = false.

Reply:
Oh you right, I missed that. 

Do you think we should remove the functionality to skip attributes with the `NewAttributeValueEmpty()` until we have a need for that?

Reply:
I removed it for now and set "string" type to be expected in the helper interface. We can change it later once other attribute types are supported.